packages/aws-cdk-lib/aws-cloudwatch/lib/metric.ts

```diff
@@ -259,6 +259,33 @@ export class Metric implements IMetric {
     });
   }
 
+  /**
+ * Make a new Alarm for this metric using anomaly detection
+ *
+ * @param scope The scope in which to create the alarm
+ * @param id The ID of the alarm
+ * @param props Alarm creation properties for anomaly detection
+ * @param metric The metric to create the alarm for
+ * @returns The newly created Alarm
+ */
+  public static createAnomalyDetectionAlarmFromMetric(scope: Construct, id: string, props: CreateAnomalyDetectionAlarmProps, metric: IMetric): Alarm {
+    if (!Alarm.isAnomalyDetectionOperator(props.comparisonOperator)) {
+      throw new Error(`Invalid comparison operator for anomaly detection alarm: ${props.comparisonOperator}`);
+    }
+    const anomalyDetectionExpression = new MathExpression({
+      expression: `ANOMALY_DETECTION_BAND(m0, ${props.bounds ?? 1})`,
+      usingMetrics: { m0: metric },
+      period: props.period,
+    });
+
+    return new Alarm(scope, id, {
+      ...props,
+      metric: anomalyDetectionExpression,
+      threshold: 0, // This value will be ignored for anomaly detection alarms
+      thresholdMetricId: 'expr_1',
```

Comment:
should we expose the property to customers? Will create 2 anomaly detection alarms with the same `thresholdMetricId` cause conflicts?

Reply:
`thresholdMetricId` is per alarm and these metric id are also local to the alarm. Which mean that there is no risk of conflict.

Conflict wise, the only risk with this change is if the user defines an expression metric with sub metrics with an `m0` identifier. I think it is fine to keep the complexity as it is now.